helix-term/src/ui/editor.rs

```diff
@@ -358,7 +361,16 @@ impl EditorView {
             let line_number_text = if line == last_line && !draw_last {
                 "    ~".into()
             } else {
-                format!("{:>5}", line + 1)
+                match config.line_number {
+                    LineNumber::Absolute => format!("{:>5}", line + 1),
+                    LineNumber::Relative => {
+                        // TODO: Put char_to_line out of the loop
+                        let current_line = doc.text().char_to_line(selections.primary().anchor);
+                        let relative_line =
+                            ((current_line as isize) - (line as isize)).abs() as usize;
```

Comment:
Define a helper func, `usize -> isize -> usize` isn't valid for all values.

```rust
#[inline]
fn abs_diff(a: usize, b: usize) -> usize {
    if a > b {
      a - b  
    } else {
      b - a
    }
}
```

Maybe one day this will be in std: https://github.com/rust-lang/rust/issues/62111